src/deterministic_vi/elbo_objective.jl

```diff
@@ -173,14 +188,27 @@ function calculate_source_pixel_brightness!{NumType <: Number}(
         # end
         # For each value in 1:Ia, written this way for speed.
         @assert Ia == 2
-        for u_ind1 = 1:2, u_ind2 = 1:2
-            E_G_s.h[ids.u[u_ind1], ids.u[u_ind2]] =
-            elbo_vars.E_G_s_hsub_vec[1].u_u[u_ind1, u_ind2] +
-            elbo_vars.E_G_s_hsub_vec[2].u_u[u_ind1, u_ind2]
-
-            E_G2_s.h[ids.u[u_ind1], ids.u[u_ind2]] =
-                elbo_vars.E_G2_s_hsub_vec[1].u_u[u_ind1, u_ind2] +
-                elbo_vars.E_G2_s_hsub_vec[2].u_u[u_ind1, u_ind2]
+        @inbounds begin
+            ids_u_1 = ids.u[1]
+            ids_u_2 = ids.u[2]
+            E_G_s_hsub_vec_1_u_u = elbo_vars.E_G_s_hsub_vec[1].u_u
+            E_G_s_hsub_vec_2_u_u = elbo_vars.E_G_s_hsub_vec[2].u_u
+            E_G2_s_hsub_vec_1_u_u = elbo_vars.E_G2_s_hsub_vec[1].u_u
+            E_G2_s_hsub_vec_2_u_u = elbo_vars.E_G2_s_hsub_vec[2].u_u
+            E_G_s_h = E_G_s.h
+            E_G2_s_h = E_G2_s.h
+
+            E_G_s_h[ids_u_1, ids_u_1] = E_G_s_hsub_vec_1_u_u[1, 1] + E_G_s_hsub_vec_2_u_u[1, 1]
+            E_G2_s_h[ids_u_1, ids_u_1] = E_G2_s_hsub_vec_1_u_u[1, 1] + E_G2_s_hsub_vec_2_u_u[1, 1]
+
+            E_G_s_h[ids_u_1, ids_u_2] = E_G_s_hsub_vec_1_u_u[1, 2] + E_G_s_hsub_vec_2_u_u[1, 2]
+            E_G2_s_h[ids_u_1, ids_u_2] = E_G2_s_hsub_vec_1_u_u[1, 2] + E_G2_s_hsub_vec_2_u_u[1, 2]
+
+            E_G_s_h[ids_u_2, ids_u_1] = E_G_s_hsub_vec_1_u_u[2, 1] + E_G_s_hsub_vec_2_u_u[2, 1]
+            E_G2_s_h[ids_u_2, ids_u_1] = E_G2_s_hsub_vec_1_u_u[2, 1] + E_G2_s_hsub_vec_2_u_u[2, 1]
+
+            E_G_s_h[ids_u_2, ids_u_2] = E_G_s_hsub_vec_1_u_u[2, 2] + E_G_s_hsub_vec_2_u_u[2, 2]
+            E_G2_s_h[ids_u_2, ids_u_2] = E_G2_s_hsub_vec_1_u_u[2, 2] + E_G2_s_hsub_vec_2_u_u[2, 2]
```

Comment:
We're going to be increasing `Ia` to `4` soon. We need to include "quasar" and "off" modes in addition to "star" and "galaxy". How much of speedup do we get from unrolling the loop?